src/ai/ai_scanner.cpp

```diff
@@ -125,14 +126,43 @@ AIInfo *AIScannerInfo::FindInfo(const char *nameParam, int versionParam, bool fo
 		if (this->info_list.find(ai_name_tmp) != this->info_list.end()) return static_cast<AIInfo *>(this->info_list[ai_name_tmp]);
 	}
 
-	/* See if there is a compatible AI which goes by that name, with the highest
-	 *  version which allows loading the requested version */
-	ScriptInfoList::iterator it = this->info_list.begin();
-	for (; it != this->info_list.end(); it++) {
-		AIInfo *i = static_cast<AIInfo *>((*it).second);
-		if (strcasecmp(ai_name, i->GetName()) == 0 && i->CanLoadFromVersion(versionParam) && (version == -1 || i->GetVersion() > version)) {
-			version = (*it).second->GetVersion();
-			info = i;
+	if (versionParam != -2) {
```

Comment:
Not a fan of this "magic" -2 value. What's wrong with using versionParam "as is" ? It's documented as "The version of the game script, or -1 if you want the latest"